crowbar_framework/app/models/service_object.rb

```diff
@@ -257,16 +257,19 @@ def remove_pending_elements(bc, inst, elements)
     end
   end
 
+  def restore_node_to_ready(node)
+    node.crowbar['state'] = 'ready'
+    node.crowbar['state_owner'] = ""
+    node.save
+  end
+
   def restore_to_ready(nodes)
     with_lock "BA-LOCK" do
       nodes.each do |node_name|
         node = NodeObject.find_node_by_name(node_name)
-        next if node.nil?
-
+        next if node.nil? || node.crowbar['state'] == 'crowbar-upgrade'
```

Comment:
Some clarifying comment on why the crowbar-upgrade state needs to be handled differently here could be helpful. I already forgot about that once, with really bad results :)